src/collector.rs

```diff
@@ -294,9 +294,15 @@ impl Collector {
             name: l.key().to_owned(),
             value: l.value().to_owned(),
         });
-        self.default_dimensions()
+        let all_dims: BTreeMap<_, _> = self
+            .default_dimensions()
             .chain(dimensions_from_keys)
+            .map(|d| (d.name.clone(), d))
+            .collect();
```

Comment:
Nit: Would be faster to collect into a `Vec`, (stable) sort and dedup it by dimension name https://doc.rust-lang.org/std/vec/struct.Vec.html#method.dedup .

Reply:
I tried that, but it was tricky to get {sort,dedup}_by_key to work with references, so it became way more ugly. I think I hit this one https://github.com/rust-lang/rust/issues/34162

Reply:
`sort_by`/`dedup_by` should work in that case.

Reply:
Nit: This only need to be sorted and deduped if there are extra dimensions passed